pkg/yurthub/locallb/locallb.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2024 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package locallb
+
+import (
+	"github.com/openyurtio/openyurt/cmd/yurthub/app/config"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+type locallbManager struct {
+	cloudIP         string
+	apiserverIPs    []string
+	iptablesManager *IptablesManager
+}
+
+func NewLocalLBManager(cfg *config.YurtHubConfiguration, informerFactory informers.SharedInformerFactory) (*locallbManager, error) {
+	m := &locallbManager{
+		cloudIP:         cfg.CloudProviderIP,
+		apiserverIPs:    cfg.APIServerIPs,
+		iptablesManager: NewIptablesManager(),
+	}
+	m.iptablesManager.addIptablesRules(m.cloudIP, m.apiserverIPs)
+
+	podInformer := informerFactory.Core().V1().Pods()
```

Comment:
It is not a good idea to list/watch pod for monitoring kube-apiserver IPs, because we want to narrow the permission of informer. 

maybe we can use an endpoints(like tenant-kas-svc) for storing the address of kube-apiserver, so we only need to list/watch this endpoints for getting address of kube-apiserver.

Reply:
done.